### PR TITLE
receive proxy protocol from LoadBalancer or others

### PR DESCRIPTION
**Pull request by heat1024**
This PR allows pftp to handle proxy protocols.

The source IP of the proxy protocol received from the LB, will be assigned to the sourceAddress for proxy protocol data to origin.

### CODE REVIEW

**Comment by pyama86** [comment]
great! Can you add tests?

File: pftp/client_handler.go [hunk 1]
```diff
@@ -271,3 +273,21 @@ func (c *clientHandler) parseLine(line string) {
 		c.param = params[1]
 	}
 }
+
+func (c *clientHandler) getSourceIPFromProxyHeader(line string) error {
```
Comment (pyama86) [inline pftp/client_handler.go]: If this is not used for other processing, methodization is unnecessary

**Comment by heat1024** [comment]
I just added unit test for handle proxy commands!

File: pftp/client_handler_test.go [hunk 2]
```diff
@@ -117,6 +118,52 @@ func Test_clientHandler_handleCommand(t *testing.T) {
 				line: "user pftp",
 			},
 		},
+		{
+			name: "proxy_invalid_proxyheader",
```
Comment (pyama86) [inline pftp/client_handler_test.go]: sounds good!
The abnormal tests are well written and feel good.

File: pftp/client_handler_test.go [hunk 3]
```diff
@@ -139,8 +186,10 @@ func Test_clientHandler_handleCommand(t *testing.T) {
 			)
 
 			got := clientHandler.handleCommand(tt.args.line)
-			if (got != nil && tt.wantR == nil) || (tt.wantR != nil && (got.code != tt.wantR.code || got.msg != tt.wantR.msg)) {
+			if (got != nil && tt.wantR == nil) || (tt.wantR != nil && (got.code != tt.wantR.code || got.msg != tt.wantR.msg || got.err.Error() != tt.wantR.err.Error())) {
 				t.Errorf("clientHandler.handleCommand() = %v, want %v", got, tt.wantR)
+			} else if tt.name == "proxy_ok" && clientHandler.sourceIP != "192.168.10.1:12345" {
```
Comment (pyama86) [inline pftp/client_handler_test.go]: I want to do like this.
```golang
} else if tt.name == "proxy_ok" && clientHandler.sourceIP != tt.want.SrcIP {
```

Comment (heat1024) [reply pftp/client_handler_test.go]: Yes, I agreed!

**Comment by pyama86** [comment]
great! thanks!

**Pull request merged**